### PR TITLE
add revert reasons

### PR DESCRIPTION
**Pull request by MrChico**

### CODE REVIEW

**Review by axic — APPROVED** [review]

File: deposit_contract.sol [hunk 1]
```diff
@@ -110,7 +110,7 @@ contract DepositContract is IDepositContract {
             sha256(abi.encodePacked(amount, bytes24(0), signature_root))
         ));
         // Verify computed and expected deposit data roots match
-        require(node == deposit_data_root);
+        require(node == deposit_data_root, "DepositContract: given node does not match computed deposit_data_root");
```
Comment (axic) [inline deposit_contract.sol]: Tried to rephrase this:
```suggestion
        require(node == deposit_data_root, "DepositContract: reconstructed DepositData does not match supplied deposit_data_root");
```

File: deposit_contract.sol [hunk 2]
```diff
@@ -76,18 +76,18 @@ contract DepositContract is IDepositContract {
...
-        require(pubkey.length == PUBKEY_LENGTH);
-        require(withdrawal_credentials.length == WITHDRAWAL_CREDENTIALS_LENGTH);
-        require(signature.length == SIGNATURE_LENGTH);
+        require(pubkey.length == PUBKEY_LENGTH, "DepositContract: invalid pubkey length");
+        require(withdrawal_credentials.length == WITHDRAWAL_CREDENTIALS_LENGTH, "DepositContract: invalid withdrawal_credentials length");
+        require(signature.length == SIGNATURE_LENGTH, "DepositContract: invalid signature length");
 
         // Avoid overflowing the Merkle tree (and prevent edge case in computing `branch`)
-        require(deposit_count < MAX_DEPOSIT_COUNT);
+        require(deposit_count < MAX_DEPOSIT_COUNT, "DepositContract: merkle tree full");
```
Comment (axic) [inline deposit_contract.sol:84]:
```suggestion
        require(deposit_count < MAX_DEPOSIT_COUNT, "DepositContract: merkle tree is full");
```

Sorry, shouldn't this have a verb?

**Comment by axic** [comment]
@chriseth you may want to check this

File: deposit_contract.sol (hunk 2 above)
Comment (MrChico) [reply deposit_contract.sol:84]: Either way

File: deposit_contract.sol [hunk 3]
```diff
@@ -110,7 +110,7 @@ contract DepositContract is IDepositContract {
             sha256(abi.encodePacked(amount, bytes24(0), signature_root))
         ));
         // Verify computed and expected deposit data roots match
-        require(node == deposit_data_root);
+        require(node == deposit_data_root, "DepositContract: reconstructed DepositData does not match supplied deposit_data_root");
```
Comment (chriseth) [inline deposit_contract.sol:113]: Maybe too long?

Comment (chriseth) [reply deposit_contract.sol:113]: I think the message just needs to be unique, you cannot grasp the full meaning without looking at the source anyway.

**Pull request merged**